br/pkg/checkpoint/checkpoint.go

```diff
@@ -344,6 +344,11 @@ func (r *CheckpointRunner) startCheckpointRunner(ctx context.Context, wg *sync.W
 	return errCh
 }
 
+func (r *CheckpointRunner) sendError(err error) {
+	log.Error("stop checkpoint runner due to error", zap.Error(err))
+	r.errCh <- err
```

Comment:
For farther preventing blocking, perhaps we can `select` here:
```go
select {
  case r.errCh <-err:
  default:
    log.Error("errCh blocks", logutil.ShortErr(err))
}

Reply:
done